redux-app/src/components/ModalEdit.js

```diff
@@ -0,0 +1,120 @@
+                            <Row>
+                                <Col md="3">
+                                    <Form.Group controlId="id">
+                                        <Form.Label>ID</Form.Label>
+                                        <Form.Control name="id"
+                                            type="text" required disabled
+                                            defaultValue={this.props.id}>
+                                        </Form.Control>
+                                    </Form.Group>
+                                </Col>
+
+                                <Col md="9">
+                                    <Form.Group controlId="descricao">
+                                        <Form.Label>Description</Form.Label>
+                                        <Form.Control name="descricao"
+                                            type="text" required
+                                            defaultValue={this.props.descricao}>
+                                        </Form.Control>
+                                    </Form.Group>
+                                </Col>
+
+                                <Col md="6">
+                                    <Form.Group controlId="quantidade">
+                                        <Form.Label>Quantity</Form.Label>
+                                        <Form.Control name="quantidade"
+                                            type="number" required
+                                            defaultValue={this.props.quantidade}>
+                                        </Form.Control>
+                                    </Form.Group>
+                                </Col>
+
+                                <Col md="6">
+                                    <Form.Group controlId="valor">
+                                        <Form.Label>Price</Form.Label>
+                                        <Form.Control name="valor"
+                                            type="number" required
+                                            defaultValue={this.props.valor}>
+                                        </Form.Control>
+                                    </Form.Group>
+                                </Col>
+                            </Row>
+
+                            <Form.Group>
+                                <Button variant="success" type="submit">Save</Button>
+                            </Form.Group>
+
+                        </Form>
+
+                    </div>
+                </Modal.Body>
+                <Modal.Footer>
+                    <Button onClick={this.props.onHide}>Close</Button>
+                </Modal.Footer>
+            </Modal>);
+    }
+}
+
+const mapDispatchToProps = (dispatch) => {
+    return {
+        changeList: (list) => {
+            dispatch({
+                type: "CHANGE_LIST",
+                payload: list,
+                key: 'lista'
```

Comment:
Try to define actions is some separate files for actions or something similar to be more maintanable